python/ray/train/session.py

```diff
@@ -278,14 +279,15 @@ def shutdown_session():
 
 
 @PublicAPI(stability="beta")
-def get_dataset_shard(dataset_name: Optional[str] = None) -> Optional[RayDataset]:
+def get_dataset_shard(
+    dataset_name: Optional[str] = None,
+) -> Optional[Union[RayDataset, DatasetPipeline]]:
```

Comment:
The RayDataset is already `Union[Dataset, DatasetPipeline]`, so we can leave as just `RayDataset` or explicitly as `Union[Dataset, DatasetPipeline]`, but we don't need both.

Reply:
I see, I found it confusing so would prefer to change it to the Union type for explicitness's sake.